mutable-copy-ksp/src/main/kotlin/fp/serrano/transformative/MutableCopyProcessor.kt

```diff
@@ -0,0 +1,91 @@
+package fp.serrano.transformative
+
+import com.google.devtools.ksp.processing.*
+import com.google.devtools.ksp.symbol.*
+import com.squareup.kotlinpoet.*
+import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
+import com.squareup.kotlinpoet.ksp.*
+
+class MutableCopyProcessor(private val codegen: CodeGenerator, private val logger: KSPLogger) : SymbolProcessor {
+  companion object {
+    const val ANNOTATION_NAME = "fp.serrano.MutableCopy"
+  }
+
+  override fun process(resolver: Resolver): List<KSAnnotated> {
+    resolver
+      .getSymbolsWithAnnotation(ANNOTATION_NAME)
+      .filterIsInstance<KSClassDeclaration>()
+      .forEach(::processClass)
+    return emptyList()
+  }
+
+  private fun processClass(klass: KSClassDeclaration) {
+    if (Modifier.DATA !in klass.modifiers || klass.primaryConstructor == null) {
+      logger.error(klass.notDataClassErrorMessage, klass)
+      return
+    }
+
+    // TODO: check for mutability and fail?
+
+    val packageName = klass.packageName.asString()
+    val targetTypeName = klass.simpleName.asString()
+    val mutableTypeName = "Mutable$targetTypeName"
+    val properties = klass.getAllProperties()
+    val typeParameters = klass.typeParameters
+    val typeParamResolver = typeParameters.toTypeParameterResolver()
+    val typeVariables = typeParameters.map { it.toTypeVariableName() }
+    val targetClassName = ClassName(packageName, targetTypeName).parameterizedWhenNotEmpty(typeVariables)
+    val mutableClassName = ClassName(packageName, mutableTypeName)
+    buildFile(packageName = packageName, fileName = mutableTypeName) {
+      addClass(mutableClassName) {
+        addTypeVariables(typeVariables)
+        primaryConstructor {
+          properties.forEach { property ->
+            addParameter(property.asParameterSpec(typeParamResolver))
+            addProperty(property.asPropertySpec(typeParamResolver) {
+              mutable(true).initializer(property.simpleName.asString())
+            })
+          }
+        }
+      }
+      val mutableParameterized = mutableClassName.parameterizedWhenNotEmpty(typeVariables)
+      addFunction(
+        name = "copy",
+        receiver = targetClassName,
+        returns = targetClassName,
+        typeVariables = typeVariables,
+      ) {
+        addParameter(name = "block", type = LambdaTypeName.get(receiver = mutableParameterized, returnType = UNIT))
```

Comment:
This should be annotated with [`@DslMarker`](https://kotlinlang.org/docs/type-safe-builders.html#scope-control-dslmarker) to both (1) make the compiler better aware of the scope, and (2) make IntelliJ show some elements as part of the language.

Reply:
Can you make this a separate issue? I need to get my head around this first 😅